### PR TITLE
Remove master/worker java opts from monitor java opts

### PR DESCRIPTION
**Pull request by humengyu2012**
### What changes are proposed in this pull request?

1. Remove master/worker java opts from monitor java opts;
2. Add default value for monitor java opts.

### Why are the changes needed?

https://github.com/Alluxio/alluxio/issues/16657
https://github.com/Alluxio/alluxio/issues/10958
https://github.com/Alluxio/alluxio/issues/15168

### Does this PR introduce any user facing changes?
no

### CODE REVIEW

**Comment by LuQQiu** [comment]
@humengyu2012 can you help sign the contribution license agreement (CLA)? Otherwise we do not have the permission to merge the PR, appreciate your help

File: bin/alluxio-monitor.sh [hunk 1]
```diff
@@ -53,13 +53,19 @@ get_env() {
...
   ALLUXIO_TASK_LOG="${ALLUXIO_LOGS_DIR}/task.log"
 
-  # Remove the remote debug configuration to avoid the error: "transport error 20: bind failed: Address already in use."
-  # See https://github.com/Alluxio/alluxio/issues/10958
-  # See https://github.com/Alluxio/alluxio/issues/15168
-  ALLUXIO_MASTER_MONITOR_JAVA_OPTS=$(echo ${ALLUXIO_MASTER_JAVA_OPTS} | sed 's/^-agentlib:jdwp=transport=dt_socket.*address=[0-9]*//' | sed 's/-Dcom.sun.management.jmxremote.port=[0-9]*//')
-  ALLUXIO_WORKER_MONITOR_JAVA_OPTS=$(echo ${ALLUXIO_WORKER_JAVA_OPTS} | sed 's/^-agentlib:jdwp=transport=dt_socket.*address=[0-9]*//' | sed 's/-Dcom.sun.management.jmxremote.port=[0-9]*//')
-  ALLUXIO_JOB_MASTER_MONITOR_JAVA_OPTS=$(echo ${ALLUXIO_JOB_MASTER_JAVA_OPTS} | sed 's/^-agentlib:jdwp=transport=dt_socket.*address=[0-9]*//' | sed 's/-Dcom.sun.management.jmxremote.port=[0-9]*//')
-  ALLUXIO_JOB_WORKER_MONITOR_JAVA_OPTS=$(echo ${ALLUXIO_JOB_WORKER_JAVA_OPTS} | sed 's/^-agentlib:jdwp=transport=dt_socket.*address=[0-9]*//'| sed 's/-Dcom.sun.management.jmxremote.port=[0-9]*//')
+  ALLUXIO_MONITOR_JAVA_OPTS_DEFAULT="-Xms256M -Xmx256M"
```
Comment (LuQQiu) [inline bin/alluxio-monitor.sh:56]: Why needs the default value?

File: conf/alluxio-env.sh.template [hunk 2]
```diff
@@ -72,6 +72,22 @@
 # E.g. "-Dalluxio.user.file.writetype.default=CACHE_THROUGH"
 # ALLUXIO_USER_JAVA_OPTS
 
+# Config properties set for Alluxio shell. (Default: "-Xms256M -Xmx256M")
```
Comment (LuQQiu) [inline conf/alluxio-env.sh.template:75]: Can we give more detailed description for when this opts is needed to be set?
e.g. # Config properties for the monitoring service used by Alluxio shell to monitor the healthiness of Alluxio master process.

**Review by LuQQiu — CHANGES_REQUESTED** [review]
Thanks for fixing the issue and contributing the code back, left some minor comments, otherwise LGTM

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for the contribution

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
@humengyu2012  the change is reverted because of fierce discussion internally. 

The old approach: able to reuse the address information and may be other information set by the main java opts (e.g. ALLUXIO_MASTER_MONITOR_JAVA_OPTS can reuse the information set by `ALLUXIO_MASTER_JAVA_OPTS`). This is especially helpful in Kubernetes environment where all the Alluxio configuration is set by Java opts.
The old approach has the bugs that you try to solve here.

Your new approach:
1. Improve correctness, definitely more correct
2. but reduce ease of use especially in Kubernetes environment. User may need to copy the ALLUXIO_MASTER_JAVA_OPTS into ALLUXIO_MASTER_MONITOR_JAVA_OPTS

The solution we want is a combination of both approaches:
1. Maintain the ease of use. Do not require user to copy the java opts and double the configuration file length.
2. Do not have the identified bugs

**Comment by ssz1997** [comment]
@LuQQiu 

Launching Alluxio in k8s doesn't use the monitor as k8s has its own readiness/liveness probe.

More specifically, `entrypoint.sh` directly calls `launch-process` which doesn't involve the monitor script. `alluxio-monitor.sh` is only called when alluxio is launched by using `alluxio-start.sh` script.

**Comment by jenoudet** [comment]
According to [this post](https://stackoverflow.com/a/10958164/15160749) on Stack Overflow, options use the value specified last. We could append `-Xms256M -Xmx256M` to the current opts to override specifically the memory allocation opts.

**Pull request merged**